gef.py

```diff
@@ -1854,6 +1854,13 @@ def mprotect_asm(cls, addr, size, perm):
         return "; ".join(insns)
 
 
+class ARM_M(ARM):
+    arch = "ARM-M"
+
+    all_registers = ARM.all_registers[:-1] + ["$xpsr",]
+    flag_register = "$xpsr"
```

Comment:
The register name is case sensitive: "$xPSR"

Reply:
That's why I had to lowercase it. The documentation also only mentions `xpsr`:

> For M-profile targets (e.g. Cortex-M3), the ‘org.gnu.gdb.arm.core’ feature is replaced by ‘org.gnu.gdb.arm.m-profile’. It should contain registers ‘r0’ through ‘r13’, ‘sp’, ‘lr’, ‘pc’, and ‘xpsr’. 

https://sourceware.org/gdb/current/onlinedocs/gdb/ARM-Features.html

Which gdb version are you using? I'm using `GNU gdb (GDB) 10.2`.

Reply:
Yes, unfortunately I have seen it with both spellings. I think it either depends on the gdbserver (and black magic probe might spell it one way) or the gdb client.

We could maybe at support for flag_register being a list, and if it is, then try each one?

Reply:
Good point. I was using QEMU as gdbserver, so the (mis)spelling probably comes from there.

> We could maybe at support for flag_register being a list, and if it is, then try each one?

That may require a bit more refactoring as there are some places which use `current_arch.flag_register` directly.

Reply:
Yep, but it also might support putting CPSR, xPSR, xpsr, etc, all in ONE arch.

Reply:
That would be nice. But they aren't completely equal, that also needs to be taken care of.

Most relevant is probably the Thumb bit:
> If you compare this [xPSR] with the Current Program Status Register (CPSR) in ARM7, [...] Thumb bit (T) is moved to bit 24.

https://www.waveshare.com/w/upload/7/76/Cortex-M3-Definitive-Guide-EN.pdf

Reply:
🤕  alright.